godel_surface_detection/src/services/surface_blending_service.cpp

```diff
@@ -1003,6 +991,8 @@ void SurfaceBlendingService::visualizePaths()
   }
 
   tool_path_markers_pub_.publish(path_visualization);
+  process_planning_result_.succeeded = true;
+  process_planning_server_.setSucceeded(process_planning_result_);
```

Comment:
Have you considered lifting the action code out of `visualizePaths()` and into the handler callback? It seems a little non-intuitive to look at a visualization function and see a code related to some state which may or may not be active.